lib/exercises.js

```diff
@@ -20,10 +71,55 @@ function topKFrequentElements(list, k) {
 //   Each element can either be a ".", or a digit 1-9
 //   The same digit cannot appear twice or more in the same 
 //   row, column or 3x3 subgrid
-// Time Complexity: ?
-// Space Complexity: ?
-function validSudoku(table) {
-    throw new Error("Method hasn't been implemented yet!");
+// Time Complexity: O(n^2)
+// Space Complexity: O(n^2)
```

Comment:
These would be correct in the case that we accepted arbitrary board sizes, but since we only accept 3x3 grids, we can actually say that this takes O(1) in both time and space.